controllers/productController.js

```diff
@@ -0,0 +1,12 @@
+const productService = require('../services/productService');
+const { catchAsync } = require('../utils/error');
+
+const productSearch = catchAsync(async (req, res) => {
+  const { name } = req.query;
```

Comment:
```suggestion
  const { keyword } = req.query;
```
검색이라고 한다면, 보통 상품 이름을 포함하여, 내용, 소재 기타 등등 외로도 검색 되는 경우가 많습니다. 실제 우리 서비스에서는 상품 이름을 통해 검색한다고 하더라도 이러한 경우를 생각해보면 keyword라는 이름으로 매개변수를 받으면 어떨까요?